module/merintr/inventry.c

```diff
@@ -671,18 +672,32 @@ void InventoryLButtonDown(HWND hwnd, BOOL fDoubleClick, int x, int y, UINT keyFl
 /************************************************************************/
 /*
  * InventoryLButtonUp:  User released left mouse button on inventory box.
- */
+ *    Call InventoryMoveCurrentItem if the mouse is within the inventory.
+ */    
 void InventoryLButtonUp(HWND hwnd, int x, int y, UINT keyFlags)
 {
    int temp_x, temp_y;
    room_contents_node *r;
+   POINT mouse;
+   AREA a;
 
-   if (!InventoryReleaseCapture())
+   // Multiple Sanity Checks
+   if (!InventoryReleaseCapture()) // Check that we're still in inventory
       return;
 
-   // See if mouse pointer is in main graphics area
-   if (!MouseToRoom(&temp_x, &temp_y))
+   InventoryGetArea(&a);                  // Get inventory area
+   GetCursorPos(&mouse);                  // Get mouse position
+   ScreenToClient(cinfo->hMain, &mouse);  // Get mouse position in client coordinates
+      
+   if (!MouseToRoom(&temp_x, &temp_y))       // If the mouse pointer is NOT in the main graphics area
```

Comment:
Streamlined checks.  Removed extra checks and consolidated around IsInArea

Reply:
This one is old, and is needed.

The function InventoryLButtonUp first checks if we are in the inventory area, and if so moves items around and exits the function.

Then this old check sees if we are not in the main window area (we already established we are not in the inventory area).  It will exit if so.  The reason is that the next few lines of code check for dropping items and putting items into a container.  It should only do so if we are not in the inventory window AND we are in the main 3D window.  If we are not, then the function exits.

TLDR; this is old and needed.